openmc/data/ace.py

```diff
@@ -0,0 +1,460 @@
+"""This module is for reading ACE-format cross sections. ACE stands for "A
+Compact ENDF" format and originated from work on MCNP_. It is used in a number
+of other Monte Carlo particle transport codes.
+
+ACE-format cross sections are typically generated from ENDF_ files through a
+cross section processing program like NJOY_. The ENDF data consists of tabulated
+thermal data, ENDF/B resonance parameters, distribution parameters in the
+unresolved resonance region, and tabulated data in the fast region. After the
+ENDF data has been reconstructed and Doppler-broadened, the ACER module
+generates ACE-format cross sections.
+
+.. _MCNP: https://laws.lanl.gov/vhosts/mcnp.lanl.gov/
+.. _NJOY: http://t2.lanl.gov/codes.shtml
+.. _ENDF: http://www.nndc.bnl.gov/endf
+
+"""
+
+from pathlib import PurePath
+import struct
+import sys
+
+import numpy as np
+
+from openmc.mixin import EqualityMixin
+import openmc.checkvalue as cv
+from .data import ATOMIC_SYMBOL, gnd_name
+from .endf import ENDF_FLOAT_RE
+
+
+def get_metadata(zaid, metastable_scheme='nndc'):
+    """Return basic identifying data for a nuclide with a given ZAID.
+
+    Parameters
+    ----------
+    zaid : int
+        ZAID (1000*Z + A) obtained from a library
+    metastable_scheme : {'nndc', 'mcnp'}
+        Determine how ZAID identifiers are to be interpreted in the case of
+        a metastable nuclide. Because the normal ZAID (=1000*Z + A) does not
+        encode metastable information, different conventions are used among
+        different libraries. In MCNP libraries, the convention is to add 400
+        for a metastable nuclide except for Am242m, for which 95242 is
+        metastable and 95642 (or 1095242 in newer libraries) is the ground
+        state. For NNDC libraries, ZAID is given as 1000*Z + A + 100*m.
+
+    Returns
+    -------
+    name : str
+        Name of the table
+    element : str
+        The atomic symbol of the isotope in the table; e.g., Zr.
+    Z : int
+        Number of protons in the nucleus
+    mass_number : int
+        Number of nucleons in the nucleus
+    metastable : int
+        Metastable state of the nucleus. A value of zero indicates ground state.
+
+    """
+
+    cv.check_type('zaid', zaid, int)
+    cv.check_value('metastable_scheme', metastable_scheme, ['nndc', 'mcnp'])
+
+    Z = zaid // 1000
+    mass_number = zaid % 1000
+
+    if metastable_scheme == 'mcnp':
+        if zaid > 1000000:
```

Comment:
Can `1000000` be moved to a variable? This is extremely easy to get wrong due to a typo / missing or adding an extra 0.